### ISSUE TITLE
eids permissions

### ISSUE THREAD

**Issue by bretg** [prebid/prebid-server#1621]
The Prebid Server side of https://github.com/prebid/Prebid.js/issues/5814 -- we need to support permissioning of eids data.

1. PBS could receive ext.prebid.data.eidPermissions which is expected to be an array of objects
```
{
    ext: {
        prebid: {
            data: {
                eidpermissions: [   // prebid server will use this to filter user.ext.eids
                   {"source": "sharedid.org", "bidders": ["*"]},  // * is the default
                   {"source": "neustar.biz", "bidders": ["bidderB"]},
                   {"source": "id5-sync.com", "bidders": ["bidderA","bidderC"]}
                ]
            }
        }
    }
}
```
2. If eidPermissions is not an array, ignore it and emit a 400 error with a reasonable description in the body
3. If there's no "source" attribute in the object in each array entry, emit a 400 error with a reasonable description in the body
4. If there's no "bidders" attribute in the object in each array entry, emit a 400 error with a reasonable description in the body
5. If the value of the "bidder" attribute is not an array of strings, emit a 400 error with a reasonable description in the body
6. If one of the bidder values is an asterisk, it means this source should be passed to all bidders
7.  If a bidder value is an unrecognized biddercode, emit a 400 error with a reasonable description in the body
8. If there are no valid biddercodes in the array, emit a 400 error with a reasonable description in the body
9. If the same source is listed more than once in the array, emit a 400 error with a reasonable description in the body
10. If there are any valid eidPermissions entries, PBS-core needs to handle them by filtering user.ext.eids before passing to each bidder. Loop through user.ext.eids
    1. If user.ext.eids.source doesn't exist in eidPermissions, all bidders can receive it
    2. If user.ext.eids.source exists in eidPermissions and the bidder value is "*", all bidders can receive it
    3. If user.ext.eids.source exists in eidPermissions with specific bidders, only those bidders can receive that entry

**Comment by SyntaxNode** [comment]
> If a bidder value is an unrecognized biddercode, drop that biddercode and emit a warning

PBS currently returns an error for an unrecognized bidder code and we did the same for the CCPA no sale bidders array. The idea being the PBS will "fail fast and loud" at the sign of trouble and an unrecognized bidder code is either an invalid request setup or a missing alias.

Should we be consistent and return an error in this case too?

**Comment by bretg** [comment]
I guess as long as we yell about it in the debug output I can tolerate hard failures for all of these validations. It's a new feature. I've had trouble lately with people getting confused about when PBS drops bidders or requests for TCF2 because our debug logging isn't good enough.

**Comment by SyntaxNode** [comment]
Cool. I'm working on this now for PBS-Go.

**Comment by SyntaxNode** [comment]
Implemented In PBS-Go 0.143.0.

**Comment by bretg** [comment]
Released with PBS-Java 1.60
